src/Microsoft.AspNetCore.WebUtilities/FormDataMultipartSection.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.WebUtilities
+{
+    public class FormDataMultipartSection
+    {
+        private Lazy<string> bodyReader;
+        private ContentDispositionHeaderValue _contentDispositionHeader;
+
+        private FormDataMultipartSection(MultipartSection section)
+        {
+            Section = section;
+            _contentDispositionHeader = ContentDispositionHeaderValue.Parse(section.ContentDisposition);
+
+            bodyReader = new Lazy<string>(() =>
+            {
+                var readTask = Section.ReadAsStringAsync();
+                readTask.Wait();
```

Comment:
Why do we want to block? Probably create an async method instead? (ReadAsStringAsync)